src/passa/projects.py

```diff
@@ -0,0 +1,145 @@
+import io
```

Comment:
Did we decide that this ultimately will live in pipenv? Of all of the constructs this is the one that is pipenv specific most of all and up to this point I didn’t feel like we made all of pipenv from scratch

Reply:
Pipenv should implement its own version of the project class, unrelated to this. The code just doesn’t make sense without this. I feel I am always passing a bunch of flags around to do things; it’s better for the implementation to have this class.

Reply:
(See how everything in CLI is so straightforward with this class. Otherwise they will be a bunch of spaghetti like the `cli.py` in master.)

Reply:
To be fair the one in master is actually directly taken from a branch I was working on in `resolver.py` in pipenv :p so that kind of makes sense. 

I sort of have mixed feelings, I think resolution should be an atomic operation that needs to be handed a pipfile and an environment to resolve in -- why does it need to have a project? just take a set of requirements, produce a lockfile, etc.  Anything beyond that probably belongs in pipenv